sos/upload/targets/__init__.py

```diff
@@ -497,7 +595,22 @@ def upload_sftp(self, user=None, password=None, user_dir=None):
 
         # need to strip the protocol prefix here
         sftp_url = self.get_upload_url().replace('sftp://', '')
-        sftp_cmd = f"sftp -oStrictHostKeyChecking=no {user}@{sftp_url}"
+        if self.use_proxy_server:
+            proxy_template, success = check_netcat_syntax()
+            if success:
+                # Final ProxyCommand to use in your SFTP execution
+                proxy_cmd = (proxy_template.
+                             replace("PROXY_IP", self.get_proxy_host()).
+                             replace("PROXY_PORT", self.get_proxy_port()).
+                             replace("PROXY_USER", self.get_proxy_user()).
+                             replace("PROXY_PASS", self.get_proxy_password()))
```

Comment:
David, we are not leaking the proxy password here or in any of the other changes to, say, sos logs or similar, right?